### PR TITLE
postNewUserMessageWithPubSub() now returns a promise on user_message_new

### PR DESCRIPTION
**Pull request by lasryaric**
`postUserMessageWithPubSub()` now returns a promise that gets resolved when we receive the `user_message_new` event.
The return type associated with the Promise is a `UserMessageType`.

### CODE REVIEW

File: front/pages/api/v1/w/[wId]/assistant/conversation/index.ts [hunk 1]
```diff
@@ -42,12 +42,28 @@ async function handler(
     });
   }
 
+  const w = await Workspace.findOne({
```
Comment (spolu) [inline front/pages/api/v1/w/[wId]/assistant/conversation/index.ts:45]: Please look at other route implementation

```
const owner = auth.workspace();
  if (!owner) {
    return apiError(req, res, {
      status_code: 404,
      api_error: {
        type: "app_not_found",
        message: "The app you're trying to run was not found",
      },
    });
  }
  ```

**Review by spolu — CHANGES_REQUESTED** [review]

File: front/pages/api/v1/w/[wId]/assistant/conversation/index.ts (hunk 1 above)
Comment (lasryaric) [reply front/pages/api/v1/w/[wId]/assistant/conversation/index.ts:45]: In my brain owner means a user. Anyway not important.

**Comment by lasryaric** [comment]
addressed all your comments @spolu

**Comment by spolu** [comment]
:+1:

**Pull request merged**